### PR TITLE
Fix an issue with datetime column in Rule Engine

### PR DESCRIPTION
**Pull request by sajjad-moradi**
## Description
With the fix, Rule Engine now works with both `time` and `dateTime` types. Previously it used to only work for time column.

## Testing Done
- added a unit test for one of the rules to have dateTime column instead of time column
- locally changed the inputs of all the rules in the unit tests to have dateTime column instead of time column and they all passed. Didn't want to duplicate all the unit tests - one for time column and one for dateTime column - so I didn't include them in the PR. One unit test with dateTime column should suffice.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java [hunk 1]
```diff
@@ -181,15 +182,19 @@ private void validateQueries() {
     invalidQueries.forEach(_queryWeightMap::remove);
   }
 
-  public void registerColnameFieldType() { // create a map from colname to data type
+  public void registerColNameFieldType() { // create a map from col name to data type
```
Comment (siddharthteotia) [inline pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java:185]: Not related to the changes in this PR but can we do the following cleanup ?
- Make this private. It is not used elsewhere.
- Move the javadoc to one line before.

Comment (sajjad-moradi) [reply pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java]: Sure. Done :)

Comment (siddharthteotia) [reply pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java]: still shows public ?

Comment (sajjad-moradi) [reply pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java]: 😅 
Done now!

**Review by siddharthteotia — APPROVED** [review]

**Pull request merged**